### PR TITLE
Color variables for scarves/beanie fix

### PR DESCRIPTION
**Pull request by Niknakflak**
- removes minor iconbloat from scarves and colors them via the variables
- minor beanie fixes/organization
- Code by JJRcop: "Mob overlays for ties on humans and corgis now inherit color and alpha to make this functional"
- Turns like 40 icons into 3

### CODE REVIEW

**Comment by Cheridan** [comment]
awesome

File: code/datums/dog_fashion.dm [hunk 1]
```diff
@@ -10,6 +10,8 @@
 	// sprite that the associated item uses
 	var/icon_file
 	var/icon_state
+	var/alpha
+	var/color
```
Comment (Cheridan) [inline code/datums/dog_fashion.dm]: imo change these var names to prevent confusion with inbuilt atom alpha and color vars.

Comment (Iamgoofball) [inline code/datums/dog_fashion.dm]: actually how does this compile BYOND should be throwing a fit due to reserved var names

Comment (Niknakflak) [inline code/datums/dog_fashion.dm]: JRrcop code, he said he tested, I dunno.

Comment (JJRcop) [inline code/datums/dog_fashion.dm]: it works you know, i don't play around with my testing. but @Iamgoofball it works for the same reason icon_state works, this is a datum.

Comment (Iamgoofball) [inline code/datums/dog_fashion.dm]: h u h

Comment (Niknakflak) [inline code/datums/dog_fashion.dm]: so do we have to change them

Comment (JJRcop) [inline code/datums/dog_fashion.dm]: Niknakflak#5

**Comment by JJRcop** [comment]
sorry

**Comment by Niknakflak** [comment]
squishy

File: code/modules/clothing/under/ties.dm [hunk 2]
```diff
@@ -214,55 +214,55 @@
...
-	item_color = "yellowscarf"
+	icon_state = "scarf"
+	color = "#E0C14F" //Yellow
 
 /obj/item/clothing/tie/scarf/orange
 	name = "orange scarf"
-	icon_state = "orangescarf"
-	item_color = "orangescarf"
+	icon_state = "scarf"
+	color = "#C67A4B" //orange
```
Comment (AnturK) [inline code/modules/clothing/under/ties.dm]: This would be funnier if purple was capitalised.

**Pull request merged**